File: go/1000-7.go
package main

import (
	"fmt"
	"time"
)

func main() {
	delay := 50 * time.Millisecond
	i := 0
	n := 1000

	for i < 143 {
		time.Sleep(delay)
		i++

		if n <= 6 {
			fmt.Println("Я гуль!")
			return
		}

		fmt.Printf("%d - %d = %d\n", n, 7, n-7)
		n -= 7
	}
}

File: ruby/1000-7.rb
delay = 0.05
i = 0
n = 1000

while i < 143
  if n <= 6
	puts 'Я гуль!'
	break
  end

  puts "#{n} - 7 = #{n - 7}"
  n -= 7

  sleep delay
  i += 1
end